main.tf

```diff
@@ -208,7 +208,31 @@ data "aws_iam_policy_document" "resource" {
 }
 
 resource "aws_ecr_repository_policy" "name" {
-  for_each   = toset(local.ecr_need_policy && module.this.enabled ? local.image_names : [])
+  for_each   = toset(local.ecr_need_policy && module.this.enabled && !var.only_repository_policy ? local.image_names : [])
   repository = aws_ecr_repository.name[each.value].name
   policy     = join("", data.aws_iam_policy_document.resource[*].json)
 }
+
+resource "aws_ecr_repository_policy" "permissions_only_name" {
+  for_each   = toset(local.ecr_need_policy && module.this.enabled && var.only_repository_policy ? local.image_names : [])
+  repository = each.value
+  policy     = join("", data.aws_iam_policy_document.resource[*].json)
+}
```

Comment:
It's possible I could unify these 2 policies, and just always use the data in `local.image_names`. It should always match the values in the created repositories anyway.

Reply:
It'd be nice if we didn't have to maintain two separate resources for each 'mode'